### PR TITLE
Simplify making information_schame tables

### PR DESCRIPTION
**Pull request by lewiszlw**
## Which issue does this PR close?

Simplify making information_schame tables and avoid potential bug that someone forgets adding logic for new table here.

### CODE REVIEW

File: datafusion/core/src/catalog/information_schema.rs [hunk 1]
```diff
@@ -225,34 +213,31 @@ impl InformationSchemaConfig {
...
+        } else if name.eq_ignore_ascii_case(VIEWS) {
             Arc::new(InformationSchemaViews::new(config))
-        } else if name.eq_ignore_ascii_case("df_settings") {
+        } else if name.eq_ignore_ascii_case(DF_SETTINGS) {
             Arc::new(InformationSchemaDfSettings::new(config))
-        } else if name.eq_ignore_ascii_case("schemata") {
+        } else if name.eq_ignore_ascii_case(SCHEMATA) {
             Arc::new(InformationSchemata::new(config))
         } else {
             return Ok(None);
```
Comment (Weijun-H) [inline datafusion/core/src/catalog/information_schema.rs:243]: nit:

```suggestion
    let table: Arc<dyn PartitionStream> = match name.to_ascii_lowercase().as_str() {
        TABLES => Arc::new(InformationSchemaTables::new(config)),
        COLUMNS => Arc::new(InformationSchemaColumns::new(config)),
        VIEWS => Arc::new(InformationSchemaViews::new(config)),
        DF_SETTINGS => Arc::new(InformationSchemaDfSettings::new(config)),
        SCHEMATA => Arc::new(InformationSchemata::new(config)),
        _ => return Ok(None),
    };
```

Comment (lewiszlw) [reply datafusion/core/src/catalog/information_schema.rs:243]: Looks cleaner!

Comment (lewiszlw) [reply datafusion/core/src/catalog/information_schema.rs:243]: Cannot commit this change due to "Applying suggestions on deleted lines is currently not supported.". I'll change it myself.

File: datafusion/core/src/catalog/information_schema.rs [hunk 2]
```diff
@@ -264,10 +244,7 @@ impl SchemaProvider for InformationSchemaProvider {
     }
 
     fn table_exist(&self, name: &str) -> bool {
-        matches!(
-            name.to_ascii_lowercase().as_str(),
-            TABLES | VIEWS | COLUMNS | SCHEMATA
-        )
+        INFORMATION_SCHEMA_TABLES.contains(&name)
```
Comment (comphead) [inline datafusion/core/src/catalog/information_schema.rs:247]: we probably want to preserve case insensitive behavior like it was before

File: datafusion/core/src/catalog/information_schema.rs [hunk 3]
```diff
@@ -225,37 +213,29 @@ impl InformationSchemaConfig {
...
+        INFORMATION_SCHEMA_TABLES
+            .iter()
+            .map(|t| t.to_string())
+            .collect()
     }
 
     async fn table(
         &self,
         name: &str,
     ) -> Result<Option<Arc<dyn TableProvider>>, DataFusionError> {
```
Comment (comphead) [inline datafusion/core/src/catalog/information_schema.rs:230]: Not related to this PR but do we need Result in return 🤔

**Review by comphead — COMMENTED** [review]
Thanks @lewiszlw please take care on case insensitive comment

File: datafusion/core/src/catalog/information_schema.rs (hunk 2 above)
Comment (lewiszlw) [reply datafusion/core/src/catalog/information_schema.rs]: oh, thanks.

**Comment by lewiszlw** [comment]
> I am surprised that this pr passed the tests, how about adding some tests for it?

I added a test in slt.

File: datafusion/core/src/catalog/information_schema.rs (hunk 3 above)
Comment (alamb) [reply datafusion/core/src/catalog/information_schema.rs:230]: I think it is related to the fact this is in the implementation of the `SchemaProvider` trait

**Review by alamb — APPROVED** [review]
Looks great to me -- thank you @lewiszlw 

Also thank you @Weijun-H 🙏

**Pull request merged**